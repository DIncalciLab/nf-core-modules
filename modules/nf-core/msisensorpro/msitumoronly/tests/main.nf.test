nextflow_process {

    name "Test Process MSISENSORPRO_MSITUMORONLY"
    script "../main.nf"
    process "MSISENSORPRO_MSITUMORONLY"

    tag "modules"
    tag "modules_nfcore"
    tag "msisensorpro"
    tag "msisensorpro/msitumoronly"
    tag "msisensorpro/scan"

    setup {
        run("MSISENSORPRO_SCAN") {
            script "../../scan/main.nf"
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                // file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                    ]
                """
            }
        }
    }

    test("human - cram") {

        when {
            process {
                """
                input[0] = [// meta map
                        [ id:'test'],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram'], checkIfExists: true),
                        // file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram_crai'], checkIfExists: true),
                        //  file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                        []
                    ]
                input[1] = [ [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                    ]
                input[2] = MSISENSORPRO_SCAN.out.list
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("human - cram - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [// meta map
                        [ id:'test'],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram_crai'], checkIfExists: true),
                        []
                    ]
                input[1] = [ [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                    ]
                input[2] = MSISENSORPRO_SCAN.out.list
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
