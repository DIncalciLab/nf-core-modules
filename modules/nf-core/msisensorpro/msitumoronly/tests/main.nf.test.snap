{
    "human - cram - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "output_all": [
                    
                ],
                "output_dis": [
                    
                ],
                "output_report": [
                    
                ],
                "output_unstable": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T11:32:08.214756"
    },
    "human - cram": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "output_all": [
                    
                ],
                "output_dis": [
                    
                ],
                "output_report": [
                    
                ],
                "output_unstable": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T11:31:59.733765"
    }
}